utils/phenotype_loading.py

```diff
@@ -641,15 +641,56 @@ def parse_first_occurrence(x):
     return mt
 
 
-def load_covid_data(all_samples_ht: hl.Table, covid_data_path: str, wave: str = '01'):
+def load_covid_data(all_samples_ht: hl.Table, covid_data_path: str, hes_main_path: str, hes_diag_path: str, death_path: str, wave: str = '01'):
     print(f'Loading COVID wave {wave}...')
+
+    #### PATH OF DATA USED IN TESTING ####
+    # def get_covid_data_path(wave: str = '20201103'):
+    #     return f'gs://ukb31063/ukb31063.covid19_test_result.{wave}.txt'
+
+    # def get_hes_main_data_path(wave: str = '20200909'):
+    #     return f'gs://ukb31063/ukb31063.hesin.{wave}.txt'
+
+    # def get_hes_diag_data_path(wave: str = '20200909'):
+    #     return f'gs://ukb31063/ukb31063.hesin_diag.{wave}.txt'
+
+    # def get_death_data_path(wave: str = '20201012'):
+    #     return f'gs://ukb31063/ukb31063.death.{wave}.txt'
+
     covid_ht = hl.import_table(covid_data_path, delimiter='\t', missing='', impute=True, key='eid')
+    hes_main_ht = hl.import_table(hes_main_path, delimiter='\t', missing='', impute=True, key='eid')
+    hes_diag_ht = hl.import_table(hes_diag_path, delimiter='\t', missing='', impute=True, key='eid')
+    death_ht = hl.import_table(death_path, delimiter='\t', missing='', impute=True, key='eid')
+
+    death_ht = death_ht.annotate(death_date=hl.experimental.strptime(death_ht.date_of_death + ' 00:00:00', '%d/%m/%Y %H:%M:%S', 'GMT'))  # Add Time Info to Death Register Data
+    hes_main_ht = hes_main_ht.annotate(diagdate=hl.or_else(hes_main_ht.epistart, hes_main_ht.admidate))
+    hes_main_ht = hes_main_ht.annotate(diag_date=hl.experimental.strptime(hes_main_ht.diagdate + ' 00:00:00', '%d/%m/%Y %H:%M:%S', 'GMT'),
+                                       admi_date=hl.experimental.strptime(hes_main_ht.admidate + ' 00:00:00', '%d/%m/%Y %H:%M:%S', 'GMT'))  # Add Time Info to HES Inpatient Data
+
+    hes_ht = hes_main_ht.key_by("eid", "ins_index").join(hes_diag_ht.key_by("eid", "ins_index"))  # Join HES Inpatient Tables
+    hes_pcr_pos = hes_ht.filter(hes_ht.diag_icd10 == "U071").key_by("eid")  # Subset Patients with Positive PCR Test Results
+    hes_other = hes_ht.filter((hes_ht.diag_icd10 != "U071") & (hes_ht.diag_icd10 != "U072")).key_by("eid")  # Subset Patients with diagnoses not related to PCR Test
+    hes_pcr_pos = hes_pcr_pos.select('diag_date', pcr_result=True)  # Select PCR-Positive Date info
+    hes_other = hes_other.select('admi_date')  # Select Admission Date info
+    hes_ht = hes_other.key_by("eid").join(hes_pcr_pos.key_by("eid"), how="outer")  # Rejoin the 2 subgroups
```

Comment:
I might be missing something about the dataset so apologies if this is the best approach, but do you need to filter and re-join? Could you get away with something like:
```
hes_ht = hes_ht.anntoate(pcr_result=hes_ht.diag_icd10 == "U071")
```
(plus whatever selected fields you need)?

Reply:
Also a minor note, is it just U071, or `(x == "U071") | (x == "U072")`?